subsys/bluetooth/host/scan.c

```diff
@@ -721,6 +721,10 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 		reset_reassembling_advertiser();
 
 cont:
+		if (buf->len < evt->length) {
+			LOG_ERR("Unexpected end of buffer");
+			break;
+		}
```

Comment:
Doesn't this check fit better right after 
```c
evt = net_buf_pull_mem(buf, sizeof(*evt));
``` 
? 

Otherwise we still attempt to parse the event if it has invalid length.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/b30b74b850ba0b8b696bfed86bf0fcd5d896fbeb/subsys/bluetooth/host/scan.c#L683-L696

@Thalley I believe that above check routine should be positioned right after the pulling of evt. Upon your confirmation, I will change it.

Reply:
I think the key thing is to do the check at least before line 707. The sooner the better though.